frontend/packages/kubevirt-plugin/src/components/create-vm/create.ts

```diff
@@ -0,0 +1,207 @@
+    );
+  const processedTemplate = await k8sCreate(ProcessedTemplatesModel, templateWrapper.asResource());
+  const vmWrapper = new VMWrapper(selectVM(processedTemplate))
+    .setNamespace(namespace)
+    .setHostname(name);
+
+  if (
+    customSource?.dataSource?.value ||
+    (!isTemplateSourceError(sourceStatus) && sourceStatus.pvc)
+  ) {
+    const rootVolume = new VolumeWrapper();
+    rootVolume.init({ name: ROOT_DISK_NAME }).setType(VolumeType.DATA_VOLUME, {
+      name: `${name}-${ROOT_DISK_NAME}`,
+    });
+
+    const rootDisk = new DiskWrapper(
+      vmWrapper.getDisks().find((d) => d.name === ROOT_DISK_NAME),
+      true,
+    ).setBootOrder(1);
+
+    let rootDataVolume;
+    let isCDRom: boolean;
+
+    if (customSource?.dataSource?.value) {
+      isCDRom = customSource.cdRom?.value;
+      rootDataVolume = getRootDataVolume({
+        name: rootVolume.getDataVolumeName(),
+        accessMode: customSource.accessMode.value,
+        cdRom: isCDRom,
+        container: customSource.container?.value,
+        pvcName: customSource.pvcName?.value,
+        pvcNamespace: customSource.pvcNamespace?.value,
+        dataSource: customSource.dataSource.value,
+        pvcSize: customSource.pvcSize?.value,
+        sizeValue: customSource.size?.value.value,
+        sizeUnit: customSource.size?.value.unit,
+        storageClass: customSource.storageClass.value,
+        url: customSource.url?.value,
+      });
+    } else if (!isTemplateSourceError(sourceStatus)) {
+      const { accessModes, resources, storageClassName } = sourceStatus.pvc.spec;
+      isCDRom = sourceStatus.isCDRom;
+      rootDataVolume = getRootDataVolume({
+        name: rootVolume.getDataVolumeName(),
+        accessMode: AccessMode.fromString(accessModes[0]),
+        cdRom: isCDRom,
+        pvcName: sourceStatus.pvc.metadata.name,
+        pvcNamespace: sourceStatus.pvc.metadata.namespace,
+        dataSource: DataSource.CLONE,
+        pvcSize: resources.requests.storage,
+        storageClass: storageClassName,
+      });
+    }
+
+    if (isCDRom) {
+      rootDisk.setType(DiskType.CDROM, { bus: rootDisk.getDiskBus() });
+      vmWrapper.prependStorage(getEmptyInstallStorage(scConfigMap, rootDisk.getDiskBus(), name));
+      vmWrapper.addAnotation(ANNOTATION_FIRST_BOOT, `${!startVM}`);
+    }
+    vmWrapper.removeStorage(ROOT_DISK_NAME);
+
+    vmWrapper.prependStorage({
+      disk: rootDisk.asResource(),
+      volume: rootVolume.asResource(),
+      dataVolume: rootDataVolume.asResource(),
+    });
+  }
+
+  if (isWindowsTemplate(template)) {
+    vmWrapper.prependStorage({
+      disk: windowsToolsStorage.disk,
+      volume: windowsToolsStorage.volume,
+    });
+  }
+
+  const settings = {
+    [VMSettingsField.NAME]: name,
+    [VMSettingsField.DESCRIPTION]: null,
+    [VMSettingsField.FLAVOR]: getFlavor(template),
+    [VMSettingsField.WORKLOAD_PROFILE]: getWorkloadProfile(template),
+    osID: getTemplateOperatingSystems([template])[0]?.id,
```

Comment:
Changed to newest os and moved the file to requests folder. I may refactor in a followup as current functions for both simple and advanced wizard heavily relies on respective types. It will be a bigger change which I want to avoid.